lib/core/src/sync/mod.rs

```diff
@@ -195,18 +213,86 @@ impl SyncService {
 
         Ok(())
     }
+
+    async fn handle_push(
+        &self,
+        record_id: &str,
+        data_id: &str,
+        record_type: RecordType,
+    ) -> Result<()> {
+        // Step 1: Get the sync state, if it exists, to compute the revision
+        let maybe_sync_state = self.persister.get_sync_state_by_record_id(&record_id)?;
+        let record_revision = maybe_sync_state
+            .as_ref()
+            .map(|s| s.record_revision)
+            .unwrap_or(0);
+        let is_local = maybe_sync_state.map(|s| s.is_local).unwrap_or(true);
+
+        // Step 2: Fetch the sync data
+        let sync_data = self.load_sync_data(data_id, record_type)?;
+
+        // Step 3: Create the record to push outwards
+        let record = Record::new(sync_data, record_revision, self.signer.clone())?;
+
+        // Step 4: Push the record
+        let req = SetRecordRequest::new(record, utils::now(), self.signer.clone())?;
+        let reply = self.client.push(req).await?;
+
+        // Step 5: Check for conflict. If present, skip and retry on the next call
+        if reply.status() == SetRecordStatus::Conflict {
+            return Err(anyhow!(
+                "Got conflict status when attempting to push record"
+            ));
+        }
+
+        // Step 6: Set/update the state revision
+        self.persister.set_sync_state(SyncState {
+            data_id: data_id.to_string(),
+            record_id: record_id.to_string(),
+            record_revision: reply.new_revision,
+            is_local,
+        })?;
+
+        Ok(())
+    }
+
+    async fn push(&self) -> Result<()> {
+        let outgoing_changes = self.persister.get_sync_outgoing_changes()?;
```

Comment:
Seems that we query only these that exists in the sync_outgoing table which seems fine to me.
just noting here that we need to populate this table for every record we want to sync on:
1. migration to this new sdk version
2. changing sync url